packages/editor/src/browser/editor-widget-factory.ts

```diff
@@ -45,7 +45,7 @@ export class EditorWidgetFactory implements WidgetFactory {
     protected async createEditor(uri: URI): Promise<EditorWidget> {
         const icon = await this.labelProvider.getIcon(uri);
         return this.editorProvider(uri).then(textEditor => {
-            const newEditor = new EditorWidget(textEditor, this.selectionService);
+            const newEditor = new EditorWidget(textEditor, this.selectionService, this.labelProvider);
```

Comment:
you can do: 
```ts
this.updateTitle(newEditor, uri);
newEditor['toDispose'].push(this.labelProvider(() => this.updateTitle(newEditor, uri)));
```

or if you don't want to access private APIs:
```ts
const toRemoveListener = this.labelProvider.onDidChange(...);
newEditor.onDispose(() => toRemoveListener.dispose());
```

Reply:
Done, using the latter suggestion.